include/boost/optional/optional.hpp

```diff
@@ -1276,6 +1276,25 @@ class optional<T&&>
 
 namespace boost {
 
+#ifndef BOOST_OPTIONAL_DETAIL_NO_RVALUE_REFERENCES
+
+template<class T>
+inline
+optional<BOOST_DEDUCED_TYPENAME boost::decay<T>::type> make_optional ( T && v  )
+{
+  return optional<BOOST_DEDUCED_TYPENAME boost::decay<T>::type>(boost::forward<T>(v));
+}
+
+// Returns optional<T>(cond,v)
+template<class T>
+inline
+optional<BOOST_DEDUCED_TYPENAME boost::decay<T>::type> make_optional ( bool cond, T && v )
+{
+  return optional<BOOST_DEDUCED_TYPENAME boost::decay<T>::type>(cond,boost::forward<T>(v));
+}
+
+#else
```

Comment:
Was it intended to remove the l-value methods with this `#else`?
I'm getting a compile error trying to make an optional from a variable, e.g. `float x = 1; make_optional<float>(x);`
(without `-DBOOST_OPTIONAL_DETAIL_NO_RVALUE_REFERENCES`)

Reply:
Function `make_optional` is not intended to be used like this, with template arguments specified explicitly. The only purpose of function `make_optional` is to spare you the trouble of specifying the type of optional value. If you want to specify it yourself, you have a simpler version:

```c++
float x = 1; boost::optional<float>(x);
```
Macro `BOOST_OPTIONAL_DETAIL_NO_RVALUE_REFERENCES` is not intended to be defined by users. The library defines it when it detects old compilers without rvalue reference support.

Reply:
Sorry I missed some detail… I was attempting to use `boost::make_optional<T>(bool, T)` to simplify the following:
```
float value;
bool valid = GetValidAndValue(&value);

// return valid ? boost::optional<float>(value) : boost::none;
return boost::make_optional<float>(valid, value);
```
True it is not a huge simplification - but should that be possible?

Reply:
Again, same response. If you can afford spelling out the type of optional value use:

```c++
return boost::optional<float>(valid, value);
```
If you want to the type of optional value to be deduced use:

```c++
return boost::make_optional(valid, value);
```

Reply:
IOW, `make_optional` is only used for type deduction.

Reply:
Ah ok, got it... thanks!